harness/tests/determined/common/experimental/test_checkpoint.py

```diff
@@ -0,0 +1,60 @@
+import os
+
+import boto3
+import moto
+import pytest
+
+from determined.common import api
+from determined.common.experimental import checkpoint
+from tests.fixtures import api_responses
+
+_MASTER = "http://localhost:8080"
+
+
+@pytest.fixture
+def standard_session() -> api.Session:
+    return api.Session(master=_MASTER, user=None, auth=None, cert=None)
+
+
+@pytest.fixture
+def sample_checkpoint(standard_session: api.Session) -> checkpoint.Checkpoint:
+    bindings_checkpoint = api_responses.sample_get_checkpoint().checkpoint
+    return checkpoint.Checkpoint._from_bindings(bindings_checkpoint, standard_session)
+
+
+# Typing directive can be removed when https://github.com/getmoto/moto/issues/4944 is resolved.
+@moto.mock_s3  # type: ignore
+def test_download_can_download_from_s3_directly(
```

Comment:
can we change the name of this to indicate that it's actually testing a specific mode of checkpoint download (DownloadMode.DIRECT)? upon first glance, i thought it was just testing that we can download from s3, didn't realize there was different modes.